apps/astarte_appengine_api/test/support/helpers/device.ex

```diff
@@ -104,7 +108,64 @@ defmodule Astarte.Helpers.Device do
     Enum.any?(result, &valid_result?(&1, interface, value))
   end
 
+  def valid_result?(result, interface, value) do
+    similar?(result, value)
+  end
```

Comment:
I would argue that there should be different tests for each of this cases, rather than trying to fit so-called similar values in a single result. We should know that e.g. some interfaces return a list and other an object, for instance. This holds for all `similar?/2` heads